test/integration/tests/create.sh

```diff
@@ -27,9 +27,20 @@ tpm2_createprimary -Q -C o -g sha1 -G rsa -c context.out
 # values.
 for gAlg in `populate_hash_algs`; do
     for GAlg in rsa keyedhash ecc aes; do
-        echo "tpm2_create -Q -C context.out -g $gAlg -G $GAlg -u key.pub \
+        echo "tpm2_create -Q -C context.out -g $gAlg -G $GAlg -u key.pub
         -r key.priv"
-        tpm2_create -Q -C context.out -g $gAlg -G $GAlg -u key.pub -r key.priv
+
+        # Some TPMs might not be able to create aes256 keys (error 0x000002c4)
```

Comment:
use getcap and look up the size of the encrypted buffer. There is a field in the output that actually relates.

Reply:
I added a function is_alg_supported based on tpm2_testparms instead. It can be used for different types of algorithms such as aes and rsa.